packages/ckeditor5-engine/src/model/schema.ts

```diff
@@ -432,7 +436,11 @@ export default class Schema extends /* #__PURE__ */ ObservableMixin() {
 	 * @param positionOrBaseElement The position or base element to which the `elementToMerge` will be merged.
 	 * @param elementToMerge The element to merge. Required if `positionOrBaseElement` is an element.
 	 */
-	public checkMerge( positionOrBaseElement: Position | Element, elementToMerge: Element ): boolean {
+	public checkMerge( positionOrBaseElement: Position | Element, elementToMerge?: Element ): boolean {
+		if ( elementToMerge && this.isLimit( elementToMerge ) ) {
+			return false;
+		}
```

Comment:
I'd move this `if` below and merge with line 475 into:

```javascript
if ( this.isLimit( positionOrBaseElement ) || this.isLimit( elementToMerge ) ) {
```

It should work, because I think that at that point we are in a place were both arguments are available and normalized to model nodes. It will be more clear to read.